app/models/user.rb

```diff
@@ -1,8 +1,12 @@
 class User < ActiveRecord::Base
   authenticates_with_sorcery!
 
-  validates :email, :password, presence: true
+  validates :email, presence: true
   validates :email, uniqueness: true, if: -> { email.present? }
+  validates :password, presence: true, on: :update, confirmation: true
+
+  before_create :setup_activation
+  after_create :send_activation_needed_email!
```

Comment:
Can we move this to the controller? Having this kind of callbacks in your model causes weird behaviour. For example, when I create an user on the terminal, I don't want to send out this email.